x-pack/plugins/canvas/public/components/sidebar/sidebar_content.tsx

```diff
@@ -43,51 +44,50 @@ const strings = {
     }),
 };
 
-const mapStateToProps = (state) => ({
-  selectedToplevelNodes: getSelectedToplevelNodes(state),
-  selectedElementId: getSelectedElementId(state),
-});
-
-const MultiElementSidebar = () => (
+const MultiElementSidebar: React.FC = () => (
   <Fragment>
     <SidebarHeader title={strings.getMultiElementSidebarTitle()} />
     <EuiSpacer />
     <MultiElementSettings />
   </Fragment>
 );
 
-const GroupedElementSidebar = () => (
+const GroupedElementSidebar: React.FC = () => (
   <Fragment>
     <SidebarHeader title={strings.getGroupedElementSidebarTitle()} groupIsSelected />
     <EuiSpacer />
     <GroupSettings />
   </Fragment>
 );
 
-const SingleElementSidebar = ({ selectedElementId }) => (
+const SingleElementSidebar: React.FC<{ selectedElementId: string | null }> = ({
+  selectedElementId,
+}) => (
   <Fragment>
     <SidebarHeader title={strings.getSingleElementSidebarTitle()} showLayerControls />
     <ElementSettings selectedElementId={selectedElementId} />
   </Fragment>
 );
 
-const branches = [
-  // multiple elements are selected
-  branch(
-    ({ selectedToplevelNodes }) => selectedToplevelNodes.length > 1,
-    renderComponent(MultiElementSidebar)
-  ),
-  // a single, grouped element is selected
-  branch(
-    ({ selectedToplevelNodes }) =>
-      selectedToplevelNodes.length === 1 && selectedToplevelNodes[0].includes('group'),
-    renderComponent(GroupedElementSidebar)
-  ),
-  // a single element is selected
-  branch(
-    ({ selectedToplevelNodes }) => selectedToplevelNodes.length === 1,
-    renderComponent(SingleElementSidebar)
-  ),
-];
+export const SidebarContent: React.FC<{ commit?: Function }> = () => {
+  const selectedToplevelNodes = useSelector<State, string[]>((state) =>
```

Comment:
nit: our pattern is to create "wrapper" components that do this kind of fetching and pass pure props to rendering components.

`sidebar_content.tsx` - thin, redux connected component
`sidebar_content.component.tsx` - agnostic rendering component

There are a number of examples, but here's a straight-forward one:

[workpad_table_tools.tsx](https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/public/components/home/my_workpads/workpad_table_tools.tsx)
[workpad_table_tools.component.tsx](https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/public/components/home/my_workpads/workpad_table_tools.component.tsx)

Reply:
Thanks. I'll have a look)